File: 1623/1623.go
func allValidTripletsThatCanReprese(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := allValidTripletsThatCanReprese(root.Left)
	r := allValidTripletsThatCanReprese(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1623/1623.cs
public class Solution {
    public int AllValidTripletsThatCanReprese(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(AllValidTripletsThatCanReprese(root.left), AllValidTripletsThatCanReprese(root.right));
    }
}